src/realm/sync/subscriptions.hpp

```diff
@@ -264,6 +264,7 @@ class MutableSubscriptionSet : public SubscriptionSet {
 
 protected:
     friend class SubscriptionStore;
+    friend class MigrationStore;
```

Comment:
Is this just so MigrationStore can call insert_sub() ? I think I made this protected because I wanted to make sure that `Subscription`'s were always created from a valid (i.e. serialized) `Query` - which I guess isn't possible in this model. So I think we could make insert_sub() public? We should add a comment though saying that this is for internal use and that SDK's accepting user-provided queries should only use insert_or_assign() ? Or at the very least we should add a comment here about why MigrationStore gets special access to this type's protected members.

Reply:
I would go with the last suggestion